### PR TITLE
Cleanup unused dependecies

### PR DESCRIPTION
**Pull request by AndreaCrotti**
- cleanup unused dependencies
- move codox and git-version plugin to dev profile

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -5,18 +5,12 @@
                  [clj-time "0.15.1"]
                  [org.clojure/core.async "0.4.490"]
                  [danlentz/clj-uuid "0.1.7"]
-                 [environ "1.1.0"]
```
Comment (creese) [inline project.clj]: Please coordinate with @cddr on https://github.com/FundingCircle/jackdaw/pull/55. We might need this.

Comment (cddr) [reply project.clj]: That PR doesn't require `environ`. It reads the environment variables via aero in https://github.com/FundingCircle/jackdaw/pull/55/files#diff-09f64f8bbdb4c2152d66c619809f4e3c

Comment (creese) [reply project.clj]: Waiting on this:
https://github.com/FundingCircle/jackdaw/pull/55/files#r254849841

Comment (AndreaCrotti) [reply project.clj]: I don't think it should be waiting for that PR, it can just be added back in that PR if @cddr switches from aero to environ.

**Pull request merged**